shared/domains/chelonia/chelonia.js

```diff
@@ -1247,7 +1247,79 @@ export default (sbp('sbp/selectors/register', {
   'chelonia/out/propDel': async function () {
 
   },
-  'chelonia/out/encryptedOrUnencryptedPubMessage': outputEncryptedOrUnencryptedPubMessage
+  'chelonia/out/encryptedOrUnencryptedPubMessage': function ({
+    contractID,
+    contractName,
+    innerSigningKeyId,
+    encryptionKeyId,
+    signingKeyId,
+    data
+  }: {
+    contractID: string,
+    contractName: string,
+    innerSigningKeyId: ?string,
+    encryptionKeyId: ?string,
+    signingKeyId: string,
+    data: Object
+  }) {
+    const serializedData = outputEncryptedOrUnencryptedMessage.call(this, {
+      contractID,
+      contractName,
+      innerSigningKeyId,
+      encryptionKeyId,
+      signingKeyId,
+      data
+    })
+    this.pubsub.pub(contractID, serializedData)
+  },
+  'chelonia/kv/set': async function (contractID: string, key: string, data: Object, {
+    contractName,
+    innerSigningKeyId,
+    encryptionKeyId,
+    signingKeyId
+  }: {
+    contractName: string,
```

Comment:
If possible, since we're already passing in the `contractID`, to avoid a potential mismatch, can you remove `contractName` as a parameter and fetch it based on the `contractID` that's passed in?